common/sgx/quote.c

```diff
@@ -529,6 +542,208 @@ oe_result_t oe_verify_quote_with_sgx_endorsements(
+        NULL, 
+        0,
+        quote,
+        (uint32_t)quote_size,
+        expiration_check_date,
+        &collateral_expiration_status,
+        &quote_verification_result,
+        &qve_report_info,
+        sizeof(qve_report_info),
+        supplemental_data,
+        MAX_SUPPLEMENTAL_DATA_SIZE,
+        &supplemental_data_size_out,
+        *(uint32_t*)(sgx_endorsements
+                        ->items[OE_SGX_ENDORSEMENT_FIELD_VERSION].data),
+        (void*)(sgx_endorsements
+                    ->items[OE_SGX_ENDORSEMENT_FIELD_TCB_INFO].data),
+        sgx_endorsements
+            ->items[OE_SGX_ENDORSEMENT_FIELD_TCB_INFO].size,
+        (void*)(sgx_endorsements
+                    ->items[OE_SGX_ENDORSEMENT_FIELD_TCB_ISSUER_CHAIN].data),
+        sgx_endorsements
+            ->items[OE_SGX_ENDORSEMENT_FIELD_TCB_ISSUER_CHAIN].size,
+        (void*)(sgx_endorsements
+                    ->items[OE_SGX_ENDORSEMENT_FIELD_CRL_PCK_CERT].data),
+        sgx_endorsements
+            ->items[OE_SGX_ENDORSEMENT_FIELD_CRL_PCK_CERT].size,
+        (void*)(sgx_endorsements
+                    ->items[OE_SGX_ENDORSEMENT_FIELD_CRL_PCK_PROC_CA].data),
+        sgx_endorsements
+            ->items[OE_SGX_ENDORSEMENT_FIELD_CRL_PCK_PROC_CA].size,
+        (void*)(sgx_endorsements
+                    ->items[OE_SGX_ENDORSEMENT_FIELD_CRL_ISSUER_CHAIN_PCK_CERT].data),
+        sgx_endorsements
+            ->items[OE_SGX_ENDORSEMENT_FIELD_CRL_ISSUER_CHAIN_PCK_CERT].size,
+        (void*)(sgx_endorsements
+                    ->items[OE_SGX_ENDORSEMENT_FIELD_QE_ID_INFO].data),
+        sgx_endorsements
+            ->items[OE_SGX_ENDORSEMENT_FIELD_QE_ID_INFO].size,
+        (void*)(sgx_endorsements
+                    ->items[OE_SGX_ENDORSEMENT_FIELD_QE_ID_ISSUER_CHAIN].data),
+        sgx_endorsements
+            ->items[OE_SGX_ENDORSEMENT_FIELD_QE_ID_ISSUER_CHAIN].size));
+
+    result = (oe_result_t)retval;
+
+    if (result != OE_PLATFORM_ERROR) {
+        OE_TRACE_INFO("SGX DCAP QvE-based quote verification is used, res: %s\n", result);
+
+        if (result != OE_OK) {
+            OE_RAISE_MSG(result,
+                "SGX QvE-based quote verification failed with error 0x%x",
+                result);
+        }
+
+        // Defense in depth
+        if (memcmp(&nonce, &qve_report_info.nonce, sizeof(sgx_nonce_t)) != 0) {
+            OE_RAISE_MSG(
+                OE_UNEXPECTED,
+                "Nonce during SGX quote verification has been tampered.\n",
+                OE_UNEXPECTED);
+        }
+
+        // Call internal API to verify QvE identity
+        OE_CHECK_MSG(oe_verify_qve_report_and_identity(
+            quote,
+            (uint32_t)quote_size,
+            &qve_report_info,
+            expiration_check_date,
+            collateral_expiration_status,
+            quote_verification_result,
+            supplemental_data,
+            supplemental_data_size_out,
+            qve_isvsvn_threshold
+        ),
```

Comment:
Looks like format-code is doing something weird here

Reply:
Will run `format-code` script.